tests/acceptance/demo-page-test.js

```diff
@@ -0,0 +1,35 @@
+import { module, test } from 'qunit';
+import { currentURL } from '@ember/test-helpers';
+import { setupApplicationTest } from 'codecrafters-frontend/tests/helpers';
+import demoPage from 'codecrafters-frontend/tests/pages/demo-page';
+
+module('Acceptance | demo page', function (hooks) {
+  setupApplicationTest(hooks);
+
+  test('it redirects from /demo to /demo/code-mirror', async function (assert) {
```

Comment:
We've found that breaking tests down to this level is usually not very useful (just adds to test times & volume of test code) unless it is mimic-ing important user flows that we explicitly want to test. 

I'd recommend just keeping a single test here and testing the basics. Our goal here should be to ensure that with the least test code we can ensure that something major isn't broken on the demo page. For example, we don't need to test whether tabs are rendered/present since we're implicitly testing clicking on them later. We also don't need a specific test for redirects as long as it is implicitly tested in another test. 

Example of a test that I think would be more appropriate here: 

```js
test('it works', async function (assert) {
  await demoPage.visit();
  assert.strictEqual(currentURL(), '/demo/code-mirror');
  assert.ok(demoPage.demoTabs.codeMirror.isPresent, 'CodeMirror tab is active by default');
  
  await demoPage.tabSwitcher.clickOnTab('DarkModeToggle');
  assert.ok(demoPage.demoTabs.darkModeToggle.component.isPresent, 'DarkModeToggle tab is active');
  
  await demoPage.tabSwitcher.clickOnTab('FileContentsCard');
  assert.ok(demoPage.demoTabs.fileContentsCard.component.isPresent, 'FileContentsCard tab is active');
  
  await demoPage.tabSwitcher.clickOnTab('CodeMirror');
  assert.ok(demoPage.demoTabs.codeMirror.component.isPresent, 'CodeMirror tab is active');
});
```

Reply:
Done! Collapsed all tests into one and it works.